tgui/packages/tgui/interfaces/BarberingMenu.js

```diff
@@ -0,0 +1,140 @@
+import { useBackend, useLocalState } from '../backend';
+import { Box, Button, Section, Image, Stack, ByondUi, Input, Icon } from '../components';
+import { Window } from '../layouts';
+
+export const BarberingMenu = (props, context) => {
+  const { data } = useBackend(context);
+  const { available_styles } = data;
+  const [searchText, setSearchText] = useLocalState(context, 'searchText', '');
+
+  const styles_keys = Object.keys(available_styles);
+
+  return (
+    <Window
+      width={700}
+      height={500}
+      title="Barber">
+
+      <Window.Content scrollable>
+        <HairOptions />
+        <hr />
+        <Stack>
+          <Stack.Item width="70%">
+            <Stack wrap="wrap" justify="space-around">
+              <Stack width="100%" inline justify="space-around">
+                <Icon name="magnifying-glass" />
+                <Stack.Item mb="10px" grow>
+                  <Input width="100%" onInput={(e, value) => setSearchText(value)} />
+                </Stack.Item>
+              </Stack>
+              <HairPreviewList search_text={searchText}
+                all_hair_names={styles_keys}
+                all_hair_styles={available_styles} />
+            </Stack>
+          </Stack.Item>
+          <Stack.Item position="fixed" width="200px">
+            <PreviewWindow />
+          </Stack.Item>
+        </Stack>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const HairPreviewList = function (props, context) {
+  const { search_text, all_hair_names, all_hair_styles } = props;
+
+  const filtered_list = all_hair_names.filter((x) => x.toLowerCase().includes(search_text.toLowerCase()));
+  return filtered_list.map((hairName) => (<HairPreview
+    key={all_hair_styles[hairName].id}
+    hair_style={all_hair_styles[hairName]}
+    hair_name={hairName} />));
+};
+
+const HairOptions = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { selected_hair_portion } = data;
+
+  return (
+    <Stack justify="space-between" fontSize="15px">
+      <Stack.Item>
+        <Button.Checkbox mx="20px" checked={selected_hair_portion === "bottom" ? 1 : 0} onClick={() => act("change_hair_portion", { "new_portion": "bottom" })}>Bottom Hair</Button.Checkbox>
+        <Button.Checkbox mx="20px" checked={selected_hair_portion === "middle" ? 1 : 0} onClick={() => act("change_hair_portion", { "new_portion": "middle" })}>Middle Hair</Button.Checkbox>
+        <Button.Checkbox mx="20px" checked={selected_hair_portion === "top" ? 1 : 0} onClick={() => act("change_hair_portion", { "new_portion": "top" })}>Top Hair</Button.Checkbox>
+      </Stack.Item>
+      <Stack.Item width="2px">
+        <Box backgroundColor="white" width="2px" height="100%" />
+      </Stack.Item>
+      <Stack.Item shrink>
+        <Button bold color="red" icon="cut" onClick={() => act("do_hair", { "style_id": null })}>Create Wig</Button>
+      </Stack.Item>
+    </Stack>
```

Comment:
Can probably be written simply using `Flex` and `Flex.Item` around each thing, rather than having one item for the buttons that have hardcoded mx, one for the Create Wig button, and one for the (unnecessary?) fake border.